### ISSUE TITLE
Contract functions that doesn't exist should trigger error

### ISSUE THREAD

**Issue by crisgarner** [ethereum-ts/TypeChain#191]
Currently in the tests, if you misstype or call a function that doesn't exists the function is assigned a type of any, meaning that it doesn't display an error until you run the tests. 

This happens because all contracts inhjerit from the class Contract which has this line:

 ```   js
export declare class Contract {
    readonly [name: string]: ContractFunction | any;
}
 ```
removing that line fixes the issue for me.

**Comment by krzkaczor** [comment]
Yes, this looks like it shouldn't be there. Which target do you use? I am not sure if we control that code.

**Comment by crisgarner** [comment]
I'm using `typechain-target-ethers`

**Comment by krzkaczor** [comment]
@quezak didn't you encounter that already?

Currently, we use `Contract` base type from ethers so we would need to extract and fix it https://github.com/ethereum-ts/TypeChain/blob/master/packages/typechain-target-ethers/lib/generation.ts#L40

**Comment by quezak** [comment]
Yeah, that's because our contract classes are just more specific type _declarations_ for the ethers normal `Contract` class -- at runtime all contracts are a `Contract` instance anyway.

I'll have to take a closer look if we can update our typings to reject other functions without changing the approach -- but we might have to add a concrete class, not just a declaration.

**Comment by krzkaczor** [comment]
This works now 🥳
